articles/getting-started-with-jwt-using-angular8-and-nodejs/index.md

```diff
@@ -0,0 +1,293 @@
+
+```javascript
+app.use(expressJWT({ secret: secret, algorithms: ['HS256']})
+    .unless( // This allows access to /token/sign without token authentication
+        { path: [
+            '/token/sign'
+        ]}
+    ));
+```
+Next, I will be testing `/path1` using Postman, with no token sent in the header.
+
+![401 Error](/engineering-education/getting-started-with-jwt-using-angular8-and-nodejs/401-unauthorized.png)
+
+The app did not allow me to access the path. Accessing the path returns a `401 Unauthorized` as expected. The code below will allow access to `/path1` if the token obtained from `/token/sign` is sent along. Add this code to `app.js`.
+```javascript
+//upon successful token authentication, access to path1 is granted
+app.get('/path1', (req, res) => {
+    res.status(200)
+        .json({
+            "success": true,
+            "msg": "Secret Access Granted"
+        });
+});
+```
+On adding the Bearer Token and accessing `/path1`, the success message is displayed as shown below:
+
+![success message](/engineering-education/getting-started-with-jwt-using-angular8-and-nodejs/success-token.png)
+
+Back to Angular, I will create a new component `home` by running the following command:
+```bash
+$ ng generate component home
+```
+Then I will modify `home.component.ts` located at `src/app` to look as the code below:
+```javascript
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styleUrls: ['./home.component.css']
+})
+export class HomeComponent implements OnInit {
+
+  readonly API_URL = 'http://localhost:3000';
+
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  ngOnInit() {
+  }
+
+signIn() {
+    this.http.get(this.API_URL + '/token/sign')
+      .subscribe(
+        (res) => {
+          console.log(res);
+          if (res['token']) {
+            localStorage.setItem('token', res['token']); //token here is stored in a local storage
+          }
+        },
+        (err) => {
+          console.log(err);
+        }
+      );    
+  }
+
+  getPath() {
+    this.http.get(this.API_URL + '/path1') //path1 is then requested    
+      .subscribe(
+        (res) => {
+          console.log(res);
+        },
+        (err) => {
+          console.log(err);
+        }
+      );       
+  }
+}
+```
+The `signIn()` function in the code above, creates a token and stores it in local storage. The `getPath()` function requests the path `path1` of our Node.js app. Then, I will write a simple Html page to test the two functions above.
+
+Open `src/app/home/home.component.html` and add the code below: 
+```html
+<h3>JSON Web Test</h3>
+
+<button (click)="signIn()">Sign In</button>
+<button (click)="getPath()">Get Path 1</button>
+```
+To run the Angular project, browse to the project folder and run the command below on a terminal.
+```bash
```

Comment:
Here, you have not shown us how to get this Html showing on the browser. Please add the necessary code, instructions, and explanation.